htmresearch/frameworks/layers/l2_l4_network_creation.py

```diff
@@ -303,12 +303,21 @@ def createMultipleL4L2Columns(network, networkConfig):
   for i in range(networkConfig["numCorticalColumns"]):
     suffixSrc = "_" + str(i)
     for j in range(networkConfig["numCorticalColumns"]):
-      if i != j:
-        suffixDest = "_" + str(j)
+      suffixDest = "_" + str(j)
+      if i < j:
+        # use delayed link for lateral connections because c_i is computed
+        # before c_j
         network.link(
-            "L2Column" + suffixSrc, "L2Column" + suffixDest,
-            "UniformLink", "",
-            srcOutput="feedForwardOutput", destInput="lateralInput")
+          "L2Column" + suffixSrc, "L2Column" + suffixDest,
+          "UniformLink", "",
+          srcOutput="feedForwardOutput", destInput="lateralInput",
+          propagationDelay=1)
+      elif i > j:
+        network.link(
+          "L2Column" + suffixSrc, "L2Column" + suffixDest,
+          "UniformLink", "",
+          srcOutput="feedForwardOutput", destInput="lateralInput",
+          propagationDelay=0)
```

Comment:
Also, there's no guarantee as to which order the columns will be called. The network API is free to call compute on regions in any order as long as they have the same phase.